src/public/js/util.js

```diff
@@ -185,13 +256,89 @@ var UTIL = (function() {
         throw new Error('Division did not happen properly');
       }
     }
+
     return remainder;
   }
 
+  // Returns evaluation in GF(256) of polynomial at value
+  // polynomial is represented as a big-endian array, IE: [1,2,3] -> x^2 + 2x + 3
+  // value is represented as an integer
+  function polynomialEval(polynomial, value) {
+    var valuePower = 1; // updated to represent value^i during loop
+    var result = 0;
+    var term;
+    var coefficient;
+    for (var i = 0; i < polynomial.length; i++) {
+      coefficient = polynomial[polynomial.length - 1 - i];
+      term = fieldMult(coefficient, valuePower);
+      result = result ^ term; // Galois extension field addition is xor with base field 2
+      valuePower = fieldMult(valuePower, value);
+    }
+
+    return result;
+  }
+
+  // Returns product of polynomial and scalar in GF(256)
+  // polynomial is represented as a big-endian array, IE: [1,2,3] -> x^2 + 2x + 3
+  // value is represented as an integer
+  function polynomialScale(polynomial, scalar) {
+    var scaledPolynomial = new Uint8Array(polynomial.length);
+    for (var i = 0; i < polynomial.length; i++) {
+      scaledPolynomial[i] = fieldMult(polynomial[i], scalar);
+    }
+    return scaledPolynomial;
+  }
+
+  function shift(msg, rot) {
+    return msg.split('').reduce(function(accumulator, letter, index) {
+      var shiftedCharCode = letter.charCodeAt() + rot;
+      var shiftedLetter = String.fromCharCode(shiftedCharCode);
+      return accumulator + shiftedLetter;
+    }, '');
+  }
+
+  // Returns concatenation of two typed arrays
+  // From http://stackoverflow.com/questions/14071463/how-can-i-merge-typedarrays-in-javascript
+  function mergeTypedArrays(arr1, arr2) {
```

Comment:
Something to be careful about with this function is that if you do pass in an empty array or not an argument arr1 or arr2, the function does not return a copy of the array you do pass in but rather the original array.

So if you do 
func(p1) {
copy = mergeTypedArray(p1);
copy[1] = 7
}
the parameter p1[1] will also be set to 7.

I doubt we will run in to this though, since I see no reason we would pass an empty array or non existent array.